server/integration/infinispan/src/main/java/org/jboss/as/clustering/infinispan/subsystem/HealthMetricsHandler.java

```diff
@@ -0,0 +1,217 @@
+        final ServiceController<?> controller = context.getServiceRegistry(false).getService(CacheContainerServiceName.CACHE_CONTAINER.getServiceName(cacheContainerName));
+        DefaultCacheContainer cacheManager = (DefaultCacheContainer) controller.getValue();
+
+        HealthMetrics metric = HealthMetrics.getMetric(metricName);
+        ModelNode result = new ModelNode();
+
+        if (metric == null) {
+            context.getFailureDescription().set(String.format("Unknown metric %s", metricName));
+        } else if (cacheManager == null) {
+            context.getFailureDescription().set(String.format("Unavailable cache container %s", metricName));
+        } else {
+            Health health = cacheManager.getHealth();
+            switch (metric) {
+                case CACHE_HEALTH:
+                    CacheHealth[] cacheHealths = health.getCacheHealth();
+                    List<String> perCacheHealth = new LinkedList<>();
+                    for (int i = 0; i < cacheHealths.length; ++i) {
+                        perCacheHealth.add(cacheHealths[i].getCacheName());
+                        perCacheHealth.add(cacheHealths[i].getStatus().toString());
+                    }
+                    result.set(toModelNodeCollection(perCacheHealth));
+                    break;
+                case FREE_MEMORY_KB:
+                    result.set(health.getHostInfo().getFreeMemoryInKb());
+                    break;
+                case TOTAL_MEMORY_KB:
+                    result.set(health.getHostInfo().getTotalMemoryKb());
+                    break;
+                case NUMBER_OF_NODES:
+                    result.set(health.getClusterHealth().getNumberOfNodes());
+                    break;
+                case CLUSTER_NAME:
+                    result.set(health.getClusterHealth().getClusterName());
+                    break;
+                case NUMBER_OF_CPUS:
+                    result.set(health.getHostInfo().getNumberOfCpus());
+                    break;
+                case CLUSTER_HEALTH:
+                    result.set(health.getClusterHealth().getHealthStatus().toString());
+                    break;
+                case LOG_TAIL:
+                    File path = new File(pathManager.resolveRelativePathEntry("server.log", ServerEnvironment.SERVER_LOG_DIR));
+                    try (BufferedReader reader = new BufferedReader(new FileReader(path))) {
+                        List<String> results = new LinkedList<>();
+                        for (String line=reader.readLine(); line != null; line=reader.readLine()) {
+                            //add new lines at the end and remove the head if needed
+                            results.add(line);
+                            if(results.size() > NUMBER_OF_LINES) {
+                                results.remove(0);
+                            }
+                        }
+                        result.set(toModelNodeCollection(results));
```

Comment:
I would like to avoid copy-pasting classes and we use pretty old Commons IO artifact (which doesn't have ReversedLinesFileReader). I created 2 JIRAs to address this problem: https://issues.jboss.org/browse/ISPN-7027 https://issues.jboss.org/browse/ISPN-7028